tasks/sep/010-op-l2-predeploy-upgrade-from-l1/VALIDATION.md

```diff
@@ -0,0 +1,204 @@
+      - The address of the Foundation Safe: `0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B`
+      - The safe hash to approve: `0x7e8055d58462ab08d75766766252966eda91b23097f8d96aca0547fe7aae078a`
+    - The using `cast index`, we can verify that:
+      ```shell
+        $ cast index address 0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B 8
+        0xc18fefc0a6b81265cf06017c3f1f91c040dc3227321d73c608cfbcf1c5253e5c
+      ```
+      and
+      ```shell
+        $ cast index bytes32 0x7e8055d58462ab08d75766766252966eda91b23097f8d96aca0547fe7aae078a 0xc18fefc0a6b81265cf06017c3f1f91c040dc3227321d73c608cfbcf1c5253e5c
+        0x66833911cd4988ff9068991368a392dfd91753075a1080eee9ac5b6bf6a4815b
+      ```
+      And so the output of the second command matches the key above.
+
+
+The only other state change are two nonce increments:
+
+- One on the Council or Foundation safe (`0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B` for Foundation and `0xf64bc17485f0B4Ea5F06A96514182FC4cB561977` for Council). If this is not decoded, it corresponds to key `0x05` on a `GnosisSafeProxy`.
+- One on the owner on the safe that sent the transaction.
+
+## Additional Checks To Perform
+
+### Cross-chain upgrade
+
+This is a two step process, we want to:
+1. Validate L1 execution completed successfully
+2. Validate L2 execution completed successfully
+
+### L1 Execution validation
+
+On the Tenderly simulation, go to the 'Events' tab and look for the `TransactionDeposited` event.
+
+Decode the `opaqueData` property using chisel from Foundry: 
+
+Open chisel and declare the function that'll decode the `opaqueData`:
+```solidity
+    function decode(bytes calldata opaqueData) public pure
+        returns (
+            uint256 _msgValue,
+            uint256 _value,
+            uint64 _gasLimit,
+            bool _isCreation,
+            bytes memory _data
+        )
+    {
+        uint256 offset = 0;
+
+        _msgValue = uint256(bytes32(opaqueData[offset:offset + 32]));
+        offset += 32;
+
+        _value = uint256(bytes32(opaqueData[offset:offset + 32]));
+        offset += 32;
+
+        _gasLimit = uint64(bytes8(opaqueData[offset:offset + 8]));
+        offset += 8;
+
+        _isCreation = bytes1(opaqueData[offset]) != 0x00;
+
+        offset += 1;
+        _data = opaqueData[offset:];
+
+        return (_msgValue, _value, _gasLimit, _isCreation, _data);
+    }
+```
+Decode the `opaqueData`.
+```solidity
+(uint256 _msgValue, uint256 _value, uint64 _gasLimit, bool _isCreation, bytes memory _data) = this.decode(hex"000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000030d400099a88ec40000000000000000000000004200000000000000000000000000000000000014000000000000000000000000c0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d30014");
```

Comment:
This way we encourage signers to copy from tenderly and verify against this file, instead of just copying from this file. (Plus this makes it more maintable for future runbooks with one less thing to change)
```suggestion
(uint256 _msgValue, uint256 _value, uint64 _gasLimit, bool _isCreation, bytes memory _data) = this.decode(hex"yourOpaqueDataHere");
```

Reply:
another nit sorry, let's also remove the leading underscores for readability